tests/v2/validation/configmaps/configmaps_test.go

```diff
@@ -0,0 +1,107 @@
+package configmaps
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+  "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+  
+)
+
+type SteveAPITestSuite struct {
+  suite.Suite
+  client            *rancher.Client
+  session           *session.Session
+  cluster           *management.Cluster
+}
+
+func (s *SteveAPITestSuite) TearDownSuite() {
+  s.session.Cleanup()
+}
+
+func (s *SteveAPITestSuite) SetupSuite() {
+  testSession := session.NewSession()
+  s.session = testSession
+
+  client, err := rancher.NewClient("", testSession)
+  require.NoError(s.T(), err)
+  s.client = client
+
+  clusterName := client.RancherConfig.ClusterName
+  require.NotEmptyf(s.T(), clusterName, "Cluster name to install should be set")
+  clusterID, err := clusters.GetClusterIDByName(s.client, clusterName)
+  require.NoError(s.T(), err, "Error getting cluster ID")
+  s.cluster, err = s.client.Management.Cluster.ByID(clusterID)
+  require.NoError(s.T(), err)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapCreate() {
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configMapSteveType)
+  
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, namespace, nil, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was created")
+  assert.Contains(s.T(), configmapObj.Name, configmapName)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapUpdate(){
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configMapSteveType)
+
+  log.Info(("Creating a configmap"))
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, namespace, nil, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Updating the configmap")
+  updatedConfigMapObj, err := updateConfigmapAnnotations(*configMapClient, configmapObj, map[string]string{annoKey: annoVal, descKey: descVal})
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was properly updated")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations, annoKey)
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations[annoKey], annoVal)
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations, descKey)
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations[descKey], descVal)
```

Comment:
Can we may be add a function like [this](https://github.com/rancher/rancher/blob/7846f19f3f2731d68101a7981c2de2cfa8fcefb8/tests/v2/validation/rbac/rbac.go#L93C1-L98C47), so we can validate all the annotations with key and value, instead of verifying each annotation and key separately. Because if we want to add additional annotations, we would have to add to validations for each of them separately. 
Let me know your thoughts.